TACC/tup-ui
# test(tup-cms): static placeholder, edu camp faq ⚠️

## Status

- [ ] Solve "Known Issues".

## Overview

Offer snippet so specific re-usable content can be placed anywhere within Content.

<details><summary>Use Cases</summary>

- The F.A.Q. on Education Camp pages: [GenCyber](https://dev.tup.tacc.utexas.edu/education/k-12-students/high-school-camps/gencyber/), [GenCyber Level UP](https://dev.tup.tacc.utexas.edu/education/k-12-students/high-school-camps/gencyber-level-up/), [Connected](https://dev.tup.tacc.utexas.edu/education/k-12-students/high-school-camps/connected/)

</details>

## Related

None.

## Changes

- **adds** a template that calls a static placeholder

## Testing

1. Login as CMS admin.
2. Add snippet.
3. Set "TEMPLATE" to `snippets/education-camp-faq.html`.
4. Save.
5. In edit mode, on any page, add Text plugin into new "Education Camp Faq" placeholder.
6. Verify that Text added appears where snippet was added.

## UI

...

## Notes

A snippet template is **not** required. Just placing this content in a snippet's "HTML" section will work. But I hesitate to showcase that power to editors and let them rely on that power, because I am not aware of the consequences.

### Known Issues

1. Placeholder does not show up in Structure mode unless:
    1. Snippet is added to Footer.
    2. Page being edited does **not** show "Content" placeholder e.g.
        - [Latest News](https://dev.tup.tacc.utexas.edu/news/latest-news/)
        - [Dashboard > Projects](https://dev.tup.tacc.utexas.edu/dashboard/projects) (or any sub section) after refreshing page
        - [Login Page](http://localhost:8000/dashboard/login?edit) while logged in  as CMS-only user, not TACC user
